tests/ui/methods.stderr

```diff
@@ -1,12 +1,20 @@
+error: methods called `new` usually return `Self`
+  --> $DIR/methods.rs:96:5
+   |
+LL | /     async fn new() -> Option<Self> {
+LL | |         None
+LL | |     }
+   | |_____^
+   |
+   = note: `-D clippy::new-ret-no-self` implied by `-D warnings`
+
 error: methods called `new` usually return `Self`
   --> $DIR/methods.rs:104:5
    |
 LL | /     fn new() -> i32 {
 LL | |         0
 LL | |     }
    | |_____^
-   |
-   = note: `-D clippy::new-ret-no-self` implied by `-D warnings`
 
```

Comment:
do we need some extra code to avoid FPs here?

Reply:
So this looks like a fp to me but haven't figured out yet how to work around that